### PR TITLE
Add real world tests

### PR DESCRIPTION
**Pull request by layomia**

### CODE REVIEW

File: src/System.Text.Json/tests/RealWorld/BlogPost.cs [hunk 1]
```diff
@@ -0,0 +1,850 @@
...
+using Xunit;
+
+namespace System.Text.Json.Tests.RealWorld.BlogPost
+{
+    public class Test : RealWorldTest
+    {
+        public override void Initialize()
+        {
+            // This payload was obtained from https://medium.com/@fagnerbrack/a-smart-programmer-understands-the-problems-worth-fixing-dcf15871f943?format=json.
+            PayloadAsString = File.ReadAllText("Resources/Payloads/BlogPost.json").StripWhitespace();
```
Comment (stephentoub) [inline src/System.Text.Json/tests/RealWorld/BlogPost.cs]: What's the license associated with it?  Are we able to just incorporate it into the repo?  Can we instead create an example that has similar characteristics but that's not a copy of code from elsewhere?

File: src/System.Text.Json/tests/RealWorld/BlogPost.cs [hunk 2]
```diff
@@ -0,0 +1,850 @@
...
+
+using System.Collections.Generic;
+using System.IO;
+using System.Text.Json.Serialization;
+using System.Text.RegularExpressions;
+using Xunit;
+
+namespace System.Text.Json.Tests.RealWorld.BlogPost
+{
+    public class Test : RealWorldTest
```
Comment (stephentoub) [inline src/System.Text.Json/tests/RealWorld/BlogPost.cs]: We're only ever going to have a single test based on incorporating content from a "BlogPost"?

File: src/System.Text.Json/tests/RealWorld/Profile.cs [hunk 3]
```diff
@@ -0,0 +1,355 @@
...
+using System.Text.Json.Serialization;
+using Xunit;
+
+namespace System.Text.Json.Tests.RealWorld.Profile
+{
+    public class Test : RealWorldTest
+    {
+        public override void Initialize()
+        {
+            // This payload was obtained with permission from Layomi Akinrinade.
```
Comment (stephentoub) [inline src/System.Text.Json/tests/RealWorld/Profile.cs]: What is it from?  Why is the "real world"?

File: src/System.Text.Json/tests/RealWorld/BlogPost.cs (hunk 1 above)
Comment (danmosemsft) [reply src/System.Text.Json/tests/RealWorld/BlogPost.cs]: Or perhaps replace all the strings with garbage.

File: src/System.Text.Json/tests/Resources/Payloads/Tweets.json [hunk 4]
```diff
@@ -0,0 +1,551 @@
+﻿{
```
Comment (danmosemsft) [inline src/System.Text.Json/tests/Resources/Payloads/Tweets.json]: Are all these licensed compatibly?

File: src/System.Text.Json/tests/Resources/Payloads/BlogPost.json [hunk 5]
```diff
@@ -0,0 +1,996 @@
+﻿{
```
Comment (danmosemsft) [inline src/System.Text.Json/tests/Resources/Payloads/BlogPost.json]: Consider moving these into https://github.com/dotnet/corefx-testdata (an established pattern) if they are particularly large. Not sure what our usual threshold is. All summing to 50K maybe?

**Comment by danmosemsft** [comment]
I don't have broader context and real world tests are good, but if these passed first time that might suggest that the code paths are already sufficiently covered.

**Comment by ericstj** [comment]
@layomia what's the state of this PR: are you planning to merge?

**Comment by maryamariyan** [comment]
cc: @layomia

**Comment by layomia** [comment]
Closing this for now, will reopen when I'm ready to check it in.

**Pull request closed without merging**